doc/devices/drivers/gpio/gpio.rst

```diff
@@ -0,0 +1,294 @@
+
+**GPIO_INT_DEBOUNCE**
+    Enable GPIO pin trigger debounce.
+
+**GPIO_INT_LEVEL**
+    Do Level trigger.
+
+**GPIO_INT_EDGE**
+    Do Edge trigger.
+
+**GPIO_INT_DOUBLE_EDGE**
+    Interrupt triggers on both rising and falling edge.
+
+
+GPIO device tree support
+************************
+
+Nodes
+=====
+
+GPIO Controller Node
+--------------------
+
+The GPIO device is defined by a GPIO controller node.
+
+Required properties are:
+
+    - gpio-controller: Indicates this device is a GPIO controller
+    - compatible:
+    - label: Should be a name string
+
+
+Interaction with the PINCTRL device driver
+==========================================
+
+To enable a pin controller to report the pins that are associated to a GPIO device
+the gpio-ranges property shall be set in the device tree. Also the label property
+shall be set and used as device name.
+
+The gpio-ranges property is of the form:
+
+.. code-block:: text
+
+    gpio-ranges = <&[pin controller] [pin number in GPIO pin number space]
+                                     [pin number in pin controller number space]
+                                     [number of pins]>,
+                                     /* ... */
+                                     ;
+
+A GPIO port definition in the device tree may look like:
+
+.. code-block:: DTS
+
+    pinctrl: pin-controller@48000000 {
+        pin-controller;
+        /* ... */
+    };
+
+    gpioa: gpio@40020000 {
+        gpio-controller;
+        compatible = "st,stm32-gpio-pinctrl";
+        #gpio-cells = <2>;
+        reg = <0x40020000 0x400>;
+        label = "STM32_GPIOA";
+        gpio-ranges = <&pinctrl GPIO_PORT_PIN0 PINCTRL_STM32_PINA0 16>;
+    };
+
+
+GPIO device driver implementation
+*********************************
+
+Driver configuration information comes from three sources:
+    - autoconf.h (Kconfig)
+    - generated_dts_board.h (device tree)
+    - soc.h
+
+Driver configuration by autoconf
+================================
+
+See :ref:`configuration` for the the full set of Kconfig symbols that are available.
+Specific to the GPIO driver are:
+
+:c:macro:`CONFIG_GPIO`
+    Enable GPIO driver.
+
+Driver configuration by device tree
+===================================
+
+The general device information for GPIO controllers is:
+
+:c:macro:`SOC_GPIO_CONTROLLER_COUNT`
+    Number of GPIO controllers that are activated. All nodes that contain a
+    gpio-controller directive are counted.
+
+:c:macro:`SOC_GPIO_CONTROLLER_0` ...
+    The label prefix of GPIO controller 0 ... (e.g. 'ST_STM32_GPIO_PINCTRL_48000000').
+    The define is generated if a gpio-controller directive is given in the node.
+
+For a GPIO controller node the following information is extracted from the device tree
+and provided in :file:`generated_dts_board.h`.
+
+**TBD**
+
+Driver configuration by soc.h
+=============================
+
+:file:`soc.h` may contain driver relevant information like e.g. HAL header files.
```

Comment:
change ``like`` to ``such as``  (**like** is for comparison, while **such as** is for examples) and delete the ``e.g.``

    information such as HAL header files.